sudoku-logic/src/main/java/am/iunetworks/training/sudoku/logic/GameBoardManipulationsHelper.java

```diff
@@ -1,33 +1,76 @@
-package am.iunetworks.training.sudoku.logic;
-
-/**
- * This class contains all board manipulations.
- *  - Rotate
- *  - Switch columns & rows
- *  - Re-arrange
- *  - others
- * Created by Gurgen Nersesyan on 01.02.2015.
- */
-public class GameBoardManipulationsHelper {
-
-    private char[] board;
-
-    public GameBoardManipulationsHelper(String board) {
-        checkIsBoardValid(board);
-        this.board = board.toCharArray();
-    }
-
-    private void checkIsBoardValid(String board) {
-        if (board == null) {
-            throw new NullPointerException("Game board can't be null");
-        }
-        if (board.length() != 81) {
-            throw new IllegalArgumentException("Game board must be 81 chars long");
-        }
-
-        if (!board.matches("^\\d{81}$")) {
-            throw new IllegalArgumentException("Game board can only have 0-9 chars");
-        }
-    }
-
-}
+package am.iunetworks.training.sudoku.logic;
+
+/**
+ * This class contains all board manipulations.
+ *  - Rotate
+ *  - Switch columns & rows
+ *  - Re-arrange
+ *  - others
+ * Created by Gurgen Nersesyan on 01.02.2015.
+ */
+public class GameBoardManipulationsHelper {
+
+    private char[] board;
+
+    public GameBoardManipulationsHelper(String board) {
+        checkIsBoardValid(board);
+        this.board = board.toCharArray();
+    }
+
+    private void checkIsBoardValid(String board) {
+        if (board == null) {
+            throw new NullPointerException("Game board can't be null");
+        }
+        if (board.length() != 81) {
+            throw new IllegalArgumentException("Game board must be 81 chars long");
+        }
+
+        if (!board.matches("^\\d{81}$")) {
+            throw new IllegalArgumentException("Game board can only have 0-9 chars");
+        }
+    }
+    
+    public GameBoardManipulationsHelper rotateBoardOn90Angle() {
+		char[] boardRotate = new char[board.length];
+		int digitPosition = board.length - 1;
+		for (int i = 1; i < 10; i++) {
+			for (int j = board.length - (10 - i); j >= 0; j -= 9) {
+				boardRotate[j] = board[digitPosition];
+				digitPosition--;
+			}
+		}
+		board = boardRotate;
+		return this;
+	}
+    
+    public GameBoardManipulationsHelper switchRow(int selectRow, int changeRow) {
+		if (selectRow < 0 || selectRow > 8 || changeRow < 0 || changeRow > 8 || selectRow == changeRow) {

```

Comment:
split if statement to 2 separate checks.
1. if (selectRow == changeRow) then return this without error
2. if selectRow  or changeRow out of range then throw IllegalArgumentException